gapir/client/connection.go

```diff
@@ -0,0 +1,262 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/gapid/core/app/auth"
+	"github.com/google/gapid/core/log"
+	replaysrv "github.com/google/gapid/gapir/replay_service"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	gapirAuthTokenMetaDataName = "gapir-auth-token"
+)
+
+type ResourceInfoList []*replaysrv.ResourceInfo
+
+func (l *ResourceInfoList) Append(resourceID string, size uint32) {
+	*l = append(*l, &replaysrv.ResourceInfo{
+		Id:   resourceID,
+		Size: size,
+	})
+}
+
+type Payload replaysrv.Payload
```

Comment:
Is there any benefit to redeclaring all these `replay_service` types?
If you want just some syntactic sugar here, you can [use type aliases](https://golang.org/doc/go1.9#language) (new in go 1.9) which does not create a new type, and so doesn't need any casting:

```
type (
	Payload   = replaysrv.Payload
	Resources = replaysrv.Resources
	...
)
```

Please check with the vetting tools to see if these should be documented.

Reply:
This is wonderful!